packages/core/src/lib/filter/index.ts

```diff
@@ -107,36 +110,87 @@ export class FilterCore extends BaseProtocol implements IBaseProtocolCore {
     );
 
     if (!res || !res.length) {
-      throw Error(
-        `No response received for request ${request.requestId}: ${res}`
-      );
+      return {
+        failure: {
+          error: ProtocolError.REMOTE_PEER_FAULT,
```

Comment:
it's more of `no response` error than `remote peer fault`

Reply:
changed that to be a `GENERIC_FAIL` as done for other `pipe` failures. we can discuss the right error code here if required